### PR TITLE
fix(parsers): register explicit filetypes unconditionally

### PR DESCRIPTION
**Pull request by clason**
@ribru17 (but proper review this time ;))

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -58,15 +58,12 @@ end
...
-    __newindex = function(table, parsername, parserconfig)
-      rawset(table, parsername, parserconfig)
-      ts.language.register(parsername, parserconfig.filetype or parsername)
-    end,
-  })
-end
+local list = setmetatable({}, {
+  __newindex = function(table, parsername, parserconfig)
+    rawset(table, parsername, parserconfig)
+    ts.language.register(parsername, parserconfig.filetype or (vim.fn.has "nvim-0.11" == 0 and parsername))
```
Comment (ribru17) [inline lua/nvim-treesitter/parsers.lua:64]:
```suggestion
    if parserconfig.filetype or vim.fn.has "nvim-0.11" == 0 then
      ts.language.register(parsername, parserconfig.filetype or parsername)
    end
```
Right now the right most expression gets evaluated as a boolean on nightly neovim, so this throws an error

**Review by ribru17 — APPROVED** [review]

**Comment by clason** [comment]
I remember now why I tried to do it like that: I didn't want to have the `has('nvim-0.11')` check inside the metatable method, for performance reasons. But I'll guess people lazy-load anyway...

**Comment by ribru17** [comment]
Interesting. Is it possible to just leave a note to remove that section after dropping 0.10 compat and then just keep it the way it was? Or are the unnecessary `register`s actually bad

**Comment by clason** [comment]
I would have just pulled the check out. The extra register leads to duplicate entries in get_lang.

**Pull request merged**